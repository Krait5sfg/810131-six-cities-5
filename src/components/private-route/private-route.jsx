import React from 'react';
import {Redirect, Route} from 'react-router-dom';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {PagePath, AuthorizationStatus} from '../../utils/const';
import {OfferPropTypes} from '../../utils/property-type';

const PrivateRoute = ({component: FavoritePage, path, exact, authorizationStatus, favoriteOffers}) => {
  return (
    <Route
      path={path}
      exact={exact}
      render={() => authorizationStatus === AuthorizationStatus.NO_AUTH
        ? <Redirect to={PagePath.LOGIN} />
        : <FavoritePage favoriteOffers={favoriteOffers} />} />
  );
};

const mapStateToProps = (({USER}) => ({
  authorizationStatus: USER.authorizationStatus,
}));

PrivateRoute.propTypes = {
  component: PropTypes.elementType.isRequired,
  path: PropTypes.string.isRequired,
  exact: PropTypes.bool.isRequired,
  authorizationStatus: PropTypes.string.isRequired,
  favoriteOffers: PropTypes.arrayOf(OfferPropTypes).isRequired,
};

export {PrivateRoute};
export default connect(mapStateToProps)(PrivateRoute);
