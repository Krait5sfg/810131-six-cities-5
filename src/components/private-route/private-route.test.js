import React from 'react';
import renderer from 'react-test-renderer';
import {PrivateRoute} from './private-route';
import {BrowserRouter} from 'react-router-dom';

describe(`PlaceCard render correctly`, () => {
  it(`PlaceCard with authorization`, () => {
    const tree = renderer.create(<BrowserRouter><PrivateRoute path={`/favorites`} exact={true} authorizationStatus={`AUTH`} onFavoriteButtonClick={() => {}} render={() => {}} /> </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it(`PlaceCard without authorization`, () => {
    const tree = renderer.create(<BrowserRouter><PrivateRoute path={`/favorites`} exact={true} authorizationStatus={`NO_AUTH`} onFavoriteButtonClick={() => {}} render={() => {}} /> </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
