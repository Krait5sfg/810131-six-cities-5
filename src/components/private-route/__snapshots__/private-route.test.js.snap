// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard render correctly PlaceCard with authorization 1`] = `" "`;

exports[`PlaceCard render correctly PlaceCard without authorization 1`] = `" "`;
